docs/dataset/formatting-your-datasets.md

```diff
@@ -0,0 +1,162 @@
+import pandas as pd
+
+df = pd.read_csv(f"s3://kolena-public-examples/coco-2014-val/raw/coco-2014-val.csv", storage_options={"anon": True})
+image_to_boxes = defaultdict(list)
+image_to_metadata = defaultdict(dict)
+
+for record in df.itertuples():
+    coords = (float(record.min_x), float(record.min_y)), (float(record.max_x), float(record.max_y))
+    bounding_box = BoundingBox(*coords)
+    image_to_boxes[record.locator].append(bounding_box)
+    metadata = {
+        "locator": record.locator,
+        "height": record.height,
+        "width": record.width,
+        "date_captured": record.date_captured,
+        "brightness": record.brightness,
+    }
+    image_to_metadata[record.locator] = metadata
+
+df_boxes = pd.DataFrame(list(image_to_boxes.items()), columns=["locator", "ground_truths"])
+df_metadata = pd.DataFrame.from_dict(image_to_metadata, orient="index").reset_index(drop=True)
+df_merged = df_metadata.merge(df_boxes, on="locator")
+
+dataframe_to_csv(df_merged, "processed.csv")
+```
+
+### Formatting results for Object Detection
+
+For Object Detection problems, model results need to have the following columns
+for the best experience. The values for each of the columns is a [`List[ScoredLabeledBoundingBox]`](../reference/annotation.md#kolena.annotation.ScoredLabeledBoundingBox)
+
+| Column Name            | Description                                         |
+|------------------------|-----------------------------------------------------|
+| matched_inference      | Inferences that were matched to a ground truth.     |
+| unmatched_inference    | Inferences that were not matched to a ground truth. |
+| unmatched_ground_truth | Inferences that were not matched to a ground truth. |
+
+These columns are used to determine `True Postitives`, `False Positives`, and `False Negatives`.
+
+We have provided an [:kolena-widget-16: Object Detection (2D) ↗](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/object_detection_2d)
+that shows how to take raw results and perform bounding box matching to produce the values mentioned above.
+
+### To use compound metrics on the fly
+
+The Kolena web application currently supports [`precision`](../metrics/precision.md),
+[`recall`](../metrics/recall.md), [`f1_score`](../metrics/f1-score.md),
+[`accuracy`](../metrics/accuracy.md), [`false_positive_rate`](../metrics/fpr.md),
+and [`true_negative_rate`](../metrics/recall.md).
+
+To leverage these, add the following columns to your CSV: `count_TP`, `count_FP`, `count_FN`, `count_TN`.
+
+### Configuring Thumbnails
+
+Add a field called thumbnail_locator to the data, where the value points to a compressed version of the data.
+This image should have the same dimensions as the original image to ensure that overlays are rendered correctly.
+This will significantly improve the performance of Studio - images load much faster when filtering, sorting or using
+the [embedding](../dataset/advanced-usage/set-up-natural-language-search.md) sort.
```

Comment:
nit: Might be worth explaining why you'd want a thumbnail before jumping into how to configure one.

Reply:
Updated